### PR TITLE
Handle purgatory assets

### LINKED ISSUE #442: Enable user to remove allocations from asset in purgatory
If user has allocations on an assets which was move to purgatory, display the asset as highlighted and enable user to remove his allocations

To do:
- [ ] Filter out all purgatory assets where user doesn't have allocations
- [ ] Highlight in purgatory asset where user has allocations and inform the user to remove his allocations

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #442.

### CODE REVIEW

**Comment by jamiehewitt15** [comment]
I'm also getting the missing date picker on this PR (using chrome): 
image

btw it works fine for me on the live site

**Comment by KatunaNorbert** [comment]
@jamiehewitt15 I propose then to not merge this one before #413 it's merged and the date picker it's solved, but we can still test the feature of this PR and approve it

**Comment by jamiehewitt15** [comment]
> @jamiehewitt15 I propose then to not merge this one before #413 it's merged and the date picker it's solved, but we can still test the feature of this PR and approve it

I'm wondering if a pull from main will fix the date picker here too

**Comment by KatunaNorbert** [comment]
Let's see

**Comment by jamiehewitt15** [comment]
Yeah, the calendar date picker is shown for me now 

image

**Comment by EnzoVezzaro** [comment]
I think I found a couple of issues.

I tried to allocate more to my purgatory asset. At first it seems to work, but the value never changes on the columns even though the "allocated amount" has actually change (which is correct).

Screenshot 2023-02-02 at 09 25 00

Another this is that my purgatory asset is not highlighted:
Screenshot 2023-02-02 at 09 21 49

**Comment by KatunaNorbert** [comment]
df-sql doesn't show your asset as in purgatory, that's why it's not highlighted by DF-WEB:
Screenshot 2023-02-03 at 11 52 09

@trizin when does df-sql update the is_purgatory field, this asset has been moved to purgatory yesterday?

**Comment by trizin** [comment]
@KatunaNorbert It should update within 30 minutes. Seems like there's a problem with staging server, let me check.

**Comment by KatunaNorbert** [comment]
Also looks like we have an issue with the Goerli subgraph, I allocate to 2 new datasets, the contract says my allocation is 8000, but in the subgraph there are only 2 allocations of 2000 each
Screenshot 2023-02-03 at 12 25 21

**Comment by EnzoVezzaro** [comment]
> 

Yeah, I thing there's an issue, I'm having this too.

I tried adding allocation to my asset, the tx was success:
Screenshot 2023-02-03 at 07 34 30

But once I refresh the page "my allocations" goes back to the initial value:
Screenshot 2023-02-03 at 08 06 37
("Allocated amount" changes correctly 25/100)

query response: 
`{
  "data": {
    "veAllocateUser": {
      "id": "0xc50ce98b13aa9782cac4af6809007b2b035e599e",
      "veAllocation": [
        {
          "allocated": "500",
          "nftAddress": "0x075b28b44bc68206b3eee96a3086dbba5694ece0",
          "chainId": "5",
          "__typename": "VeAllocation"
        },
        {
          "allocated": "500",
          "nftAddress": "0x4d6bee7f7833c6e9927e72a8eeb3ade8aca8eade",
          "chainId": "80001",
          "__typename": "VeAllocation"
        },
        {
          "allocated": "500",
          "nftAddress": "0x7e81795b55b9e97cd706178f0a8602db57d77522",
          "chainId": "80001",
          "__typename": "VeAllocation"
        }
      ],
      "__typename": "VeAllocateUser"
    }
  }
}`

**Comment by trizin** [comment]
@KatunaNorbert The purgatory data should be up to date, could you take a look?

**Comment by EnzoVezzaro** [comment]
> @KatunaNorbert The purgatory data should be up to date, could you take a look?

Great thanks, now it's working. I just have some suggestions on the UI.

Instead of having this text like this:
Screenshot 2023-02-03 at 10 30 13

can we instead do something like this one:
Screenshot 2023-02-03 at 10 33 56

To do:

- add tooltip on assets in purgatory with message: "Item in purgatory. Please remove your allocations."
- make entire row with `color: var(--brand-alert-red);`
- block user from inputing a value grater than what they have on the row. Example: if a user has already allocated 10% to an asset in purgatory, block the user if the input is greater than 10%.

**Comment by KatunaNorbert** [comment]
@EnzoVezzaro good suggestions, but we might be limited by the library or require a lot of work to get this points done and might not worth the work.
Let me see what can be done.

**Comment by EnzoVezzaro** [comment]
This looks great. Thanks for the changes. I just find another little:

- Now I have 2 assets in purgatory, but it should be only one:
Screenshot 2023-02-06 at 11 06 49

- This one is correct:
Screenshot 2023-02-06 at 11 07 00

- But this one is not in purgatory:
Screenshot 2023-02-06 at 11 06 43

If you want I can take a look at it.

**Comment by KatunaNorbert** [comment]
Yes, I hardcoded one so it would be easier to test, so now each of us can see a purgatory asset

**Comment by EnzoVezzaro** [comment]
> And blocking the user allocation it's not yet implemented

really 🥲 on my end seems to be working, I can't add more to my allocation amount. I can only decrease the amount, which is correct, or am i missing something?

**Comment by EnzoVezzaro** [comment]
> It shouldn't work only in the case where you try to allocate more than you have available

then there's something happening. In here for example I can't set more than 15 on the first entry (example: 16).

Screenshot 2023-02-06 at 11 35 10

**Comment by KatunaNorbert** [comment]
How much do you have available?

**Comment by KatunaNorbert** [comment]
Added back the test dataset.
@EnzoVezzaro was trying to find your purgatory dataset but looks like it is not present inside the list of datasets. @trizin can you check, the did is `did:op:4f8e2a408c136ae9a50768728df8b9371a97d12b32a285ae6ec03f8c920f5944`. 

More than that, looks line there isn't any purgatory asset inside the list of datasets from df-sql

**Comment by jamiehewitt15** [comment]
I'm seeing that one of the assets I have allocated towards is in purgatory, but the [asset](https://market.oceanprotocol.com/asset/did:op:4e5040015fd68a5ea61d18c471f164ab531dc6828dde4710d0cdfcdffa62761e) isn't in purgatory:

Screenshot from 2023-02-09 11-51-12

**Comment by KatunaNorbert** [comment]
I mentioned before that I'm marking the first dataset as in purgatory for testing purposes, I can easily remove if it's confusing, but It should make testing easier.

**Comment by EnzoVezzaro** [comment]
> > More than that, looks line there isn't any purgatory asset inside the list of datasets from df-sql
> 
> yeah, maybe there's an issue with df-sql. I remember the last time this happened it was because mumbai was out-of-sync. It might be the same, but on the market you can see the [asset](https://market.oceanprotocol.com/asset/did:op:4f8e2a408c136ae9a50768728df8b9371a97d12b32a285ae6ec03f8c920f5944) being in purgatory.

my asset is back:

Screenshot 2023-02-09 at 07 36 00

but unfortunately I'm having issues allocating to my assets:

https://user-images.githubusercontent.com/13741335/217803209-82524bc2-ea39-4ac6-b2ac-4ef173736146.mp4

**Comment by KatunaNorbert** [comment]
@EnzoVezzaro  I found a solution, I'm pushing the changes now

**Comment by EnzoVezzaro** [comment]
> @EnzoVezzaro I found a solution, I'm pushing the changes now

https://user-images.githubusercontent.com/13741335/218403163-30f54faf-72c5-4b95-a7eb-f391398bbb54.mp4

Great 👏🏽 it's working perfectly. I'm not completely sure why somebody would reduce it's allocation amount instead of just remove it all. Can we maybe discuss it a bit @KatunaNorbert ? do you think makes more sense to give people the freedom to do whatever they want, or maybe we can just find a way to tell the user he needs to remove all his allocation.

**Comment by jamiehewitt15** [comment]
> do you think makes more sense to give people the freedom to do whatever they want, or maybe we can just find a way to tell the user he needs to remove all his allocation.

I think we can discuss this in another issue. It would be great to get this merged seeing as it works at delivering the core functionality.

File: src/components/common/TextWithNetworkIcon.svelte
Comment (jamiehewitt15) [inline src/components/common/TextWithNetworkIcon.svelte:18]:
```suggestion
    text={"Item in purgatory. Remove all your allocations for this asset."}
```

**Review by EnzoVezzaro — DISMISSED** [review]

**Review by jamiehewitt15 — APPROVED** [review]
Looks good to me

**Review by EnzoVezzaro — APPROVED** [review]

**Pull request merged**
